Neos.Flow/Classes/Property/PropertyMapper.php

```diff
@@ -154,6 +154,12 @@ public function getMessages()
      */
     protected function doMapping($source, $targetType, PropertyMappingConfigurationInterface $configuration, &$currentPropertyPath)
     {
+        $targetTypeWithoutNull = TypeHandling::stripNullableType($targetType);
+        if ($source === null && $targetType !== $targetTypeWithoutNull) {
```

Comment:
&lt;nitpick&gt;should we maybe inline this logic to `&& TypeHandling::isNullableType($targetType)`? Would make it a bit more readable and reusable&lt;/nitpick&gt;

Reply:
Sounds reasonable.

Reply:
Caveat: We would do the stripping call twice (which is a bit costly if it ends up in the preg_replace) I'd leave it like that for now, since we're inside property mapping which might get called quite a lot and just make the intention more explicit by assigning the $type comparison to an expressive variable (like below).